mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -273,6 +273,19 @@ def upload_trace_data(self, trace_data: Dict[str, Any]) -> None:
                     headers=headers,
                     is_single=True,
                 )
+            elif cred.credential_info.type == ArtifactCredentialType.AZURE_SAS_URI:
+                block_id = base64.b64encode(uuid.uuid4().hex.encode()).decode("utf-8")
+                put_block(
+                    sas_url=signed_uri,
+                    block_id=block_id,
+                    data=temp_file.read_bytes(),
+                    headers=headers,
+                )
+                put_block_list(
+                    sas_url=signed_uri,
+                    block_list=[block_id],
+                    headers=headers,
+                )
```

Comment:
This is a simplified version of `_azure_upload_file`.

Reply:
Can we reuse `_azure_upload_file` and update it so that it accepts custom refresh logic for tracing? Copy / pasting upload logic is what led to this issue in the first place.

Reply:
Sounds good, I'll implement it.